src/ui/opengl/castleinternalsettings.pas

```diff
@@ -261,25 +289,24 @@ procedure SettingsLoad(const Container: TUIContainer; const SettingsUrl: String)
         E.AttributeSingleDef('reference_height', DefaultUIReferenceHeight);
     end;
 
-    E := SettingsDoc.DocumentElement.Child('default_font', false);
-    if E <> nil then
+    NewFonts[ftRegular] := LoadFontSettings(SettingsDoc.DocumentElement.Child('default_font', false));
```

Comment:
This creates a structure in XML file that doesn't look perfect, because it's "flat" where it could be "hierarchical":

```
<default_font_regular>
</default_font_regular>
<default_font_bold>
</default_font_bold>
...
```

I would prefer to change to code to support this:

```
<default_font>
  <regular
      url="castle-data:/MyFontFile.ttf"
      size="20"
      sizes_at_load="10 20 30"
      anti_aliased="true"
    />
  <bold
      url="castle-data:/MyFontFile-Bold.ttf"
      size="20"
      sizes_at_load="10 20 30"
      anti_aliased="true"
    />
  <italic
      url="castle-data:/MyFontFile-Italic.ttf"
      size="20"
      sizes_at_load="10 20 30"
      anti_aliased="true"
    />
  <bold_italic
      url="castle-data:/MyFontFile-BoldItalic.ttf"
      size="20"
      sizes_at_load="10 20 30"
      anti_aliased="true"
    />
</default_font>
```

The way to do it:

In `LoadFontSettings`, check not only `if FontElement <> nil then`, check `if (FontElement <> nil) and FontElement.HasAttribute('url') then` .

Look for fonts by first checking is `DefaultFontElement := SettingsDoc.DocumentElement.Child('default_font', false); if DefaultFontElement <> nil then`, and if true, get the children like `DefaultFontElement.Child('regular', false)`.